### PR TITLE
WIP - Added toggleterm, based upon curtanips work

### PR DESCRIPTION
**Pull request by G-Rowell**
Hi all,

This is some cool work that was started off by @curtainp!
So credit to them!

## improvements,
- Installed the [nvim-toggleterm.lua plugin](https://github.com/akinsho/nvim-toggleterm.lua)
- Setup various keybinds / autocommands for it
-- `<LEADER> + h` for a horizontal terminal
-- `<LEADER> + v` for a vertical terminal
-- `<LEADER> + w` for a window terminal    (this one is my fav)
- Improved the statusline hiding by adding a `chadrc.lua` table, `chadrc.ui.hidden_statusline`
-- now we can easily hide the statusline of any filetype!

## Todo
- Make the horizontal, vertical & terminals separate maybe?
-- `nvim-toggleterm.lua` does provide this functionality already, do we want to wrap it?

*tl;dr*: monkey uses terms better

### CODE REVIEW

**Comment by G-Rowell** [comment]
Working on this still,

@curtainp if you make some commits to a branch, I can pull them in here and we can work together?

**Comment by curtainp** [comment]
> Working on this still,
> 
> @curtainp if you make some commits to a branch, I can pull them in here and we can work together?

thanks your good job, we can devote into there together.

**Comment by G-Rowell** [comment]
@curtainp 

If you want, pull my branch with `git pull https://github.com/G-Rowell/NvChad/ dev-toggleTerm --rebase`

Then make your modifications, push them to your GH, then let me know & I'll pull them back into my branch/here

That way we can work together on the PR! :)

**Comment by curtainp** [comment]
> @curtainp
> 
> If you want, pull my branch with `git pull https://github.com/G-Rowell/NvChad/ dev-toggleTerm --rebase`
> 
> Then make your modifications, push them to your GH, then let me know & I'll pull them back into my branch/here
> 
> That way we can work together on the PR! :)

ok, but for your opinions yesterday, i think we can dip into them as needed for this project philosophy -- `blazing fast and less bloated.`

**Comment by G-Rowell** [comment]
Hmmm, what are you suggesting?
Not sure I understand

**Comment by curtainp** [comment]
> Hmmm, what are you suggesting?
> Not sure I understand

I plan to read neovim help doc or code, get some internal terminal features, and then https://github.com/akinsho/nvim-toggleterm.lua see how to use this features, finally, we can impl what we want according to the design philosophy.
@G-Rowell  how about you?

**Comment by G-Rowell** [comment]
So there aren't really any internal NeoVim terminal feautures that are being used really,
this is all about adding new functionality,

`nvim-toggleterm.lua` adds it's own stuff, which we would be re-creating if we made it ourselves.

Either way there is no modification to the inbuilt Terminal features, only adding on some commands on top of it

The only real NeoVim Terminal feauture the plugin uses is akin to `:Terminal`

**Comment by curtainp** [comment]
wooo, then we can make `terminal` config more reasonable. by the way, I find that use `nvim-toggleterm.lua` a little slow than `:Terminal` directly (about ~1s) in my env.

**Comment by G-Rowell** [comment]
Ummm, I'm not getting a 1 seconds delay anywhere? 
Do you know how to replicate this delay?

Also, I'm still not sure what you mean by 'make `terminal` config more reasonable'?

As either we use the plugin `nvim-toggleterm.lua` or we re-implement it's features ourselves for little benefit?

**Comment by curtainp** [comment]
> As either we use the plugin nvim-toggleterm.lua or we re-implement it's features ourselves for little benefit?
maybe we should compare it to decide which way is more faster

**Comment by G-Rowell** [comment]
@siduck76 
Aside from all bind being the same term rn, this is probably ready?

I can look into that issue over the weekend, but it's 2am here so I have to stop :laughing:

**Review by siduck76 — APPROVED** [review]

**Comment by curtainp** [comment]
> Hi all,
> 
> This is some cool work that was started off by @curtainp!
> So credit to them!
> 
> ## improvements,
> * Installed the [nvim-toggleterm.lua plugin](https://github.com/akinsho/nvim-toggleterm.lua)
> * Setup various keybinds / autocommands for it
>   -- `<LEADER> + h` for a horizontal terminal
>   -- `<LEADER> + v` for a vertical terminal
>   -- `<LEADER> + w` for a window terminal    (this one is my fav)
> * Improved the statusline hiding by adding a `chadrc.lua` table, `chadrc.ui.hidden_statusline`
>   -- now we can easily hide the statusline of any filetype!
> 
> ## Todo
> * Make the horizontal, vertical & terminals separate maybe?
>   -- `nvim-toggleterm.lua` does provide this functionality already, do we want to wrap it?
> 
> _tl;dr_: monkey uses terms better

good job!
